translate.js

```diff
@@ -148,7 +148,41 @@ function translateJsonCompilerOutput (output) {
   return ret;
 }
 
+function formatAssemblyText (asm, prefix, source) {
+  if (typeof asm === typeof '' || asm === null || asm === undefined) {
+    return prefix + asm + '\n';
+  }
+  var text = prefix + '.code\n';
+  asm['.code'].forEach(function (item, i) {
+    var v = item.value === undefined ? '' : item.value;
+    var src = '';
+    if (source !== undefined && item.begin !== undefined && item.end !== undefined) {
+      src = source.slice(item.begin, item.end).replace('\n', '\\n', 'g');
```

Comment:
What about `\t`? Perhaps rather use some escaping function?

Reply:
True, fixed. Haven't found any built in escape function for those, do you know any?